### PR TITLE
feat(loc-field): Allow custom component for clear button

### PR DESCRIPTION
**Pull request by noletorious**
This PR would allow for a custom component to replace the default Times clear button next to the location-field input.

### CODE REVIEW

File: packages/location-field/src/index.js [hunk 1]
```diff
@@ -593,7 +594,7 @@ class LocationField extends Component {
             aria-label="Clear location"
             onClick={this.onClearButtonClick}
           >
-            <Times size={13} />
+            <ClearButtonIcon />
```
Comment (grant-humphries) [inline packages/location-field/src/index.js:597]: this line needs to look like this
```js
{ClearButtonIcon}
```

The idea is that you would instantiate the Icon component before you pass it in a as prop, that gives the user of the `location-field` component the control to pass it whatever props they want to the icon component which can allow custom styling that otp-ui won't have to worry about.  You did this with the default value, so when the default is used you trying to instantiate an already instantiated componet

**Comment by grant-humphries** [comment]
Add a storybook story for that passes a custom icon to this new prop.  This will allow to verify that the new functionality is working as expected without having to plug it into an app. You should be able to get an alternate icons from the `styled-icons` package that is already a dependency for this package

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM!

**Comment by grant-humphries** [comment]
I think this is not related to this merge request because the clear button doesn't work in other stories in this section, but just an fyi in storybook under LocationField --> DesktopContext --> Selected Location & Selected Location Custom Clear the clear button doesn't clear the text.  It does work under the MobileContext section.

Great work @noletorious! @binh-dam-ibigroup will leave it to you and the IBI team to merge

**Review by grant-humphries — APPROVED** [review]

**Review by miles-grant-ibigroup — APPROVED** [review]
Looks great! Thanks for submitting.

**Pull request merged**